Solitaire/js/main.js

```diff
@@ -1,4 +1,259 @@
+            drawTable();
+            checkGameCompletion();
+        } else {
+            deckSelected = deckNum;
+            cardSelectedNumber = event.target.id;
+            event.target.classList.add("active");
+        }
+    };
+
+    function checkGameCompletion() {
+        if (playingDecks.deckFinish1.array.length === 13 && playingDecks.deckFinish2.array.length === 13 && playingDecks.deckFinish3.array.length === 13 && playingDecks.deckFinish4.array.length === 13) {
+            alert("Поздравляем Вы выиграли!");
+            resetGame();
+        }
+    };
+
+    function resetGame() {
+        cards = [];
+		for (key in playingDecks) {
+			playingDecks[key].array = [];
+		}
+        fillingArrayCards();
+        cards.sort(mixRandom);
+        fillingGameDecks();
+        drawTable();
+    };
+
+    function moveCards(secondDeckId, secondCardId, firstDeckId, firstCardId) {
+        var secondDeck;
+        var firstDeck;
+        var movingCardIndex;
+        var compareCardSuit1;
+        var compareCardValue1;
+        var compareCardSuit2;
+        var compareCardValue2 = 0;
+        var compareCardColor1;
+        var compareCardColor2;
+		
+		
+		firstDeck = playingDecks[firstDeckId].array;
+		secondDeck = playingDecks[secondDeckId].array;
+
+
+        for (var i = 0; i < firstDeck.length; i++) {
+            if (firstDeck[i].id === firstCardId) {
+                movingCardIndex = i;
+                compareCardSuit1 = firstDeck[i].suit;
+                compareCardValue1 = firstDeck[i].value;
+            }
+        }
+
+        for (var i = 0; i < secondDeck.length; i++) {
+            if (secondDeck[i].id === secondCardId) {
+                compareCardSuit2 = secondDeck[i].suit;
+                compareCardValue2 = secondDeck[i].value;
+            }
+        }
+
+        if (compareCardSuit1 === 'hearts' || compareCardSuit1 === 'diamonds') {
+            compareCardColor1 = "red";
+        } if (compareCardSuit1 === 'clovers' || compareCardSuit1 === 'spades') {
+            compareCardColor1 = "black";
+        } if (compareCardSuit2 === 'hearts' || compareCardSuit2 === 'diamonds') {
+            compareCardColor2 = "red";
+        } if (compareCardSuit2 === 'clovers' || compareCardSuit2 === 'spades') {
+            compareCardColor2 = "black";
+        }
+
+        if (secondDeckId === 'deckFinish1' || secondDeckId === 'deckFinish2' || secondDeckId === 'deckFinish3' || secondDeckId === 'deckFinish4') {
+            if (compareCardValue1 - 1 === compareCardValue2 && compareCardSuit1 === compareCardSuit2 || compareCardValue1 === 1) {
+                spliceCards(firstDeck, secondDeck, movingCardIndex);
+            } else {
+                alert("Начать стопку Дома можно только с Туза. Все карты в стопке должны быть только одной масти и в таком порядке Туз, 2,3,4,5,6,7,8,9,10, Валет, Дама, Король.");
+            }
+
+        } else if (compareCardValue1 === compareCardValue2 - 1 || secondDeckId === secondCardId) {
```

Comment:
All the conditions in this function must be refactored. To many ifs, loops and messages here.